iothub/service/src/Amqp/AmqpClientHelper.cs

```diff
@@ -130,41 +127,41 @@ public static Exception ToIotHubClientContract(Error error)
             }
             else if (error.Condition.Equals(AmqpErrorCode.MessageSizeExceeded))
             {
-                retException = new MessageTooLargeException(message);
+                retException = new IotHubServiceException(HttpStatusCode.RequestEntityTooLarge, IotHubStatusCode.MessageTooLarge, message);
             }
             else if (error.Condition.Equals(AmqpErrorCode.ResourceLimitExceeded))
             {
-                retException = new DeviceMaximumQueueDepthExceededException(message);
+                retException = new IotHubServiceException(HttpStatusCode.Forbidden, IotHubStatusCode.DeviceMaximumQueueDepthExceeded, message);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.DeviceAlreadyExists))
             {
-                retException = new DeviceAlreadyExistsException(message, (Exception)null);
+                retException = new IotHubServiceException(HttpStatusCode.Conflict, IotHubStatusCode.DeviceAlreadyExists, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.DeviceContainerThrottled))
             {
-                retException = new IotHubThrottledException(message, (Exception)null);
+                retException = new IotHubServiceException((HttpStatusCode)429, IotHubStatusCode.ThrottlingException, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.QuotaExceeded))
             {
-                retException = new QuotaExceededException(message, (Exception)null);
+                retException = new IotHubServiceException(HttpStatusCode.Forbidden, IotHubStatusCode.IotHubQuotaExceeded, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.PreconditionFailed))
             {
-                retException = new PreconditionFailedException(message, (Exception)null);
+                retException = new IotHubServiceException(HttpStatusCode.PreconditionFailed, IotHubStatusCode.PreconditionFailed, message, (Exception)null);
             }
             else if (error.Condition.Equals(IotHubAmqpErrorCode.IotHubSuspended))
             {
-                retException = new IotHubSuspendedException(message);
+                retException = new IotHubServiceException(IotHubStatusCode.HubSuspended, message);
```

Comment:
to-do: check the error code returned by the hub service while sending requests to a suspended hub.

Reply:
ideally, we should pass the 3digit ErrorCode property into IotHubServiceException ctor

Reply:
Added "IotHubSuspended = 400020" enum to IotHubErrorCode